FireboltNETSDK/Client/FireboltStreamingDataReader.cs

```diff
@@ -0,0 +1,205 @@
+        }
+
+        /// <inheritdoc/>
+        public override int GetValues(object[] values)
+        {
+            List<object?> row = _currentRow ?? new List<object?>();
+            int n = Math.Min(row.Count, values.Length);
+            for (int i = 0; i < n; i++)
+            {
+                values[i] = GetValue(i);
+            }
+            return n;
+        }
+
+        /// <inheritdoc/>
+        public override bool IsDBNull(int ordinal)
+        {
+            ValidateRowIndex();
+            return _currentRow != null && _currentRow[ordinal] == null;
+        }
+
+        /// <inheritdoc/>
+        public override bool Read()
+        {
+            if (IsStreamClosed()) return false;
+            if (_currentRowQueue.Count == 0 && !FetchRow())
+                return false;
+
+            _currentRow = _currentRowQueue.Dequeue();
+            _currentRowIndex++;
+
+            return true;
+        }
+
+        protected override List<object?>? GetRow(int ordinal)
+        {
+            if (ordinal < 0 || ordinal > _currentRow?.Count - 1)
+            {
+                throw new InvalidOperationException($"Column ${ordinal} does not exist");
+            }
+            return _currentRow;
+        }
+
+        protected override ColumnType GetColumnType(int ordinal)
+        {
+            return ColumnType.Of(TypesConverter.GetFullColumnTypeName(_metas[ordinal]));
+        }
+
+        private bool FetchRow()
+        {
+            if (IsStreamClosed()) return false;
+
+            var line = _streamReader.ReadLine();
+            if (string.IsNullOrWhiteSpace(line)) throw new FireboltException("Failed to read line from stream");
+
+            var json = JsonConvert.DeserializeObject<StreamingJsonData>(line);
+            var messageType = json?.MessageType;
+
+            switch (messageType)
+            {
+                case Data:
+                    json?.Data.ForEach(_currentRowQueue.Enqueue);
+                    return true;
+
+                case FinishSuccessfully:
+                    _endOfStream = true;
+                    break;
+
+                default:
+                    HandleError(line);
+                    break;
+            }
+
+            return false;
+        }
+
+        private void InitializeMetadata()
+        {
+            var line = _streamReader.ReadLine();
+            if (string.IsNullOrWhiteSpace(line))
+            {
+                _endOfStream = true;
+                throw new FireboltException("Failed to read line from stream");
+            }
+
+            var json = JsonConvert.DeserializeObject<StreamingJsonStart>(line);
+
+            if (json?.MessageType != Start) HandleError(line);
+            _metas = json?.Meta ?? new List<Meta>();
+        }
+
+        private void HandleError(string line)
+        {
+            _endOfStream = true;
+            var json = JsonConvert.DeserializeObject<StreamingJsonFinishError>(line);
```

Comment:
Here I believe we can also receive an error during JSON parsing. Let's also wrap it with a meaningful error message